src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,154 @@
+package duke.main;
+
+import java.time.LocalDate;
+
+import duke.command.Command;
+import duke.command.DeleteTaskCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.MakeDeadlineCommand;
+import duke.command.MakeEventCommand;
+import duke.command.MakeToDoCommand;
+import duke.command.MarkTaskCommand;
+import duke.command.ShowListCommand;
+import duke.exception.MissingArgumentException;
+import duke.exception.MissingDescriptionException;
+import duke.exception.MissingIndexException;
+import duke.exception.MissingTimeException;
+import duke.task.Deadline;
+import duke.task.Event;
+
+/**
+ * Class which parses the input by the user.
+ */
+public class Parser {
+
+    /**
+     * Parses the full command and generates a corresponding Command object which will be executed.
+     * @param fullCommand The full command to be parsed.
+     * @return A Command object which does what the command specifies when executed.
+     * @throws MissingIndexException If the index is needed but missing.
+     * @throws MissingDescriptionException If the description is needed but missing.
+     * @throws MissingTimeException If the time is needed but missing.
+     */
```

Comment:
There should be an empty line between the description and the parameter section. I noticed the same issue in several other places too.